source/developers-guide/shopware-5-upgrade-guide-for-developers/index.md

```diff
@@ -222,6 +222,33 @@ If you want to implement this feature for your own element, you only need to pro
 interface `Shopware\Components\HttpCache\CacheTimeServiceInterface` and then tag this service in the `services.xml` with
 the tag `invalidation_date_provider`. It will then be picked up and called automatically.
 
+### Cache warmer
+
+Previously, the cache warmer relied on seo urls, but that did only cover a small amount of possible urls, which can be found
+in shopware. With these changes developers doesn't have to add unnecissary new seo urls to warm them for the cache. Also,
+the performance and amount of urls were greatly improved to cover the most content of shopware by itself.
+
+By implementing `HttpCache\UrlProvider\UrlProviderInterface` to a new service with the tag `cache_warmer.url_provider`
+developers can now easily add their own url providers for the cache warmer, which will now be called by the CLI or backend
+module after only a few changes in `WarmUpHttpCacheCommand.php` for the CLI or the `httpCache` object in `multi_request_tasks.js`
```

Comment:
please separate this sentence. it is quite long and I don't really understand, what you want to say with "a few changes"